File: signature/angularjs.go
package signature

var AngularJsSignature = Signature{
	Name:        "AngularJS",
	Description: "A discontinued free and open-source JavaScript-based web framework for developing single-page applications.",
	Cpe:         "cpa:/a:angularjs:angular.js",

	DetectPattern: DetectPattern{
		Bodies: []string{
			`angular`,
		},
		URLs: []string{
			`/angular-`,
		},
	},
	VersionPattern: VersionPattern{
		Bodies: []string{
			`\* @license AngularJS v(\d+\.\d+\.\d+)`,
			`http:\/\/errors.angularjs.org\/(\d+\.\d+\.\d+(-rc\.\d+)?)`,
		},
	},
}

File: src/signatures/technologies/angularjs.ts
import type { Signature } from "../_types.js";

export const angularJsSignature: Signature = {
  name: "AngularJS",
  description:
    "AngularJS is a JavaScript-based open-source web application framework led by the Angular Team at Google.",
  cpe: "cpe:2.3:a:angularjs:angular.js:*:*:*:*:*:*:*:*",
  rule: {
    confidence: "high",
    urls: [
      "angular[.-]([\\d.]*\\d)[^/]*\\.js",
      "/([\\d.]+(?:-?rc[.\\d]*)*)/angular(?:\\.min)?\\.js",
      "(?!angular\\.io)\\bangular.{0,32}\\.js",
    ],
    bodies: ["<(?:div|html)[^>]+ng-app=", "<ng-app"],
    javascriptVariables: {
      angular: "",
      "angular.version.full": "^(.+)$",
    },
  },
};
